### PR TITLE
8325950: Make sure all files in the JDK pass jcheck

### PR DESCRIPTION
**Pull request by magicus**
Since jcheck only checks file in a commit, there is a possibility of us getting files in the repository that would not be accepted by jcheck. This can happen when extending the set of files checked by jcheck, or if jcheck changes how it checks files (perhaps due to bug fixes).

I have now run jcheck over the entire code base, and fixed a handful of issues. With these changes, jcheck accept the entire code base.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8325950](https://bugs.openjdk.org/browse/JDK-8325950): Make sure all files in the JDK pass jcheck (**Bug** - P4)

### Reviewers
 * [Phil Race](https://openjdk.org/census#prr) (@prrace - **Reviewer**) ⚠️ Review applies to [3faa912e](https://git.openjdk.org/jdk/pull/17871/files/3faa912e26d309b3317d6a6d25b084addf9e6e0d)
 * [Bradford Wetmore](https://openjdk.org/census#wetmore) (@bradfordwetmore - **Reviewer**) ⚠️ Review applies to [3faa912e](https://git.openjdk.org/jdk/pull/17871/files/3faa912e26d309b3317d6a6d25b084addf9e6e0d)
 * [Erik Joelsson](https://openjdk.org/census#erikj) (@erikj79 - **Reviewer**)
 * [Naoto Sato](https://openjdk.org/census#naoto) (@naotoj - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/17871/head:pull/17871` \
`$ git checkout pull/17871`

Update a local copy of the PR: \
`$ git checkout pull/17871` \
`$ git pull https://git.openjdk.org/jdk.git pull/17871/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 17871`

View PR using the GUI difftool: \
`$ git pr show -t 17871`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/17871.diff">https://git.openjdk.org/jdk/pull/17871.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/17871#issuecomment-1946003474)

### CODE REVIEW

**Comment by magicus** [comment]
Some general remarks: The problems in .m4 files where not properly detected and fixed when I added .m4 to the list of checked files. The properties files were recently checked by me, so these suprrised me. It turned out I had misunderstood the jcheck criteria: I thought only leading tabs were disallowed, but it is actually tabs anywhere in the file that are banned.

In general, I have replaced tab characters with spaces aligning to tab stops at 8 characters distance. 

I'll leave some remarks for individual files.

File: src/java.naming/share/classes/com/sun/jndi/ldap/jndiprovider.properties [hunk 1]
```diff
@@ -8,7 +8,7 @@ java.naming.factory.control=com.sun.jndi.ldap.ctl.ResponseControlFactory
 java.naming.factory.object=com.sun.jndi.ldap.obj.AttrsToCorba:com.sun.jndi.ldap.obj.MarshalledToObject
 
 # RemoteToAttrs: Turn RMI/JRMP and RMI/IIOP object into javaMarshalledObject or
```
Comment (magicus) [inline src/java.naming/share/classes/com/sun/jndi/ldap/jndiprovider.properties:10]: I adjusted this tab stop (with one space) so it aligned properly with the line above; I think that was the intention.

File: src/jdk.jdeps/share/classes/com/sun/tools/jdeps/resources/jdkinternals.properties [hunk 2]
```diff
@@ -38,7 +38,7 @@ java.awt.peer=Should not use. See https://bugs.openjdk.org/browse/JDK-8037739
 java.awt.dnd.peer=Should not use. See https://bugs.openjdk.org/browse/JDK-8037739
 jdk.internal.ref.Cleaner=Use java.lang.ref.PhantomReference @since 1.2 or java.lang.ref.Cleaner @since 9
 sun.awt.CausedFocusEvent=Use java.awt.event.FocusEvent::getCause @since 9
-sun.font.FontUtilities=See java.awt.Font.textRequiresLayout	@since 9
```
Comment (magicus) [inline src/jdk.jdeps/share/classes/com/sun/tools/jdeps/resources/jdkinternals.properties:41]: This tab just looked out of place, so I replaced it with a space. (Rhyming not intentional...)

File: test/hotspot/jtreg/containers/docker/JfrReporter.java [hunk 3]
```diff
@@ -51,4 +51,3 @@ public static void main(String[] args) throws Exception {
         }
     }
```
Comment (magicus) [inline test/hotspot/jtreg/containers/docker/JfrReporter.java:52]: I'm not sure how a Java file ever got into the code base with trailing spaces, but a guess is that there have been a bug in jcheck that did not properly check for trailing whitespace at the end of the file, or something like that.

File: test/jdk/java/util/Currency/currency.properties [hunk 4]
```diff
@@ -15,7 +15,7 @@ MF=MFF,555,9
 NO=EUR   ,978  ,2,  2099-01-01T00:00:00
 SB=EUR,111,2, 2099-01-01T00:00:00
 US=euR,978,2,2001-01-01T00:00:00
-ZZ	=	ZZZ	,	999	,	3
```
Comment (magicus) [inline test/jdk/java/util/Currency/currency.properties:18]: This looks weird, but so did the original code. I assume this is to clearly indicate this as a end of list marker.

Comment (erikj79) [reply test/jdk/java/util/Currency/currency.properties:18]: This looks weird indeed. Luckily it's just test code. Did you run the test after this change?

Comment (magicus) [reply test/jdk/java/util/Currency/currency.properties:18]: All the java/util/Currency tests pass. I also searched the code for "ZZ" but could not find any references in the test.

Comment (naotoj) [reply test/jdk/java/util/Currency/currency.properties:18]: Please do not replace those tabs with spaces as they are intentional for runtime to safely ignore them. I suggest replacing them with Unicode escapes (`\u000b`)

Comment (andy-goryachev-oracle) [reply test/jdk/java/util/Currency/currency.properties:18]: `\u000b` is VT   (vertical tab)  
`\u0009` or `\t` perhaps?

Comment (naotoj) [reply test/jdk/java/util/Currency/currency.properties:18]: Right. `\t` is better to avoid such a mistake.

**Review by prrace — APPROVED** [review]
desktop changes look fine to me.

**Review by bradfordwetmore — APPROVED** [review]
security properties looks ok.

File: test/jdk/java/util/Currency/currency.properties (hunk 4 above)
Comment (magicus) [reply test/jdk/java/util/Currency/currency.properties:18]: Ok, fixed.

**Review by erikj79 — APPROVED** [review]

**Review by naotoj — APPROVED** [review]
LGTM

**Pull request closed without merging**